### PR TITLE
[WIP] [Dominion] [Site] Landing Page Header Component

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Fixes #163283824

## Type of change

- [x] New feature (non-breaking change which adds functionality)

#### Design

screenshot_2019-02-10 zeplin - project 1

#### Implementation

screenshot from 2019-02-10 11-25-57

#### Design

screenshot_2019-02-10 zeplin - project

#### Implementation

???

#### Design

screenshot_2019-02-10 zeplin - project 2

#### Implementation

???

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
I think this is still WIP @nerdlet (There are merge conflicts and I think some of changes are already merged in e.g. theme/scroll to top setup).

**Comment by nerdlet** [comment]
@kilemensi This is still in [WIP] working on the mobile side of it form the designs. Fixing the conflicts shortly

File: dominion/frontend/.eslintrc.json [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+  "parser": "babel-eslint",
   "extends": ["airbnb", "plugin:prettier/recommended", "prettier/react"],
   "env": {
     "browser": true,
     "jest": true
   },
   "rules": {
+    "react/prop-types": 0,
+    "react/forbid-prop-types": 0,
+    "import/no-named-as-default": 0,
```
Comment (kilemensi) [inline dominion/frontend/.eslintrc.json]: Don't disable rules, _fix_ the issues they raise. All these are fixable in code.

File: dominion/frontend/.eslintrc.json [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
         "extensions": [".js", ".jsx"]
       }
     ]
+  },
+  "settings": {
+    "import/resolver": {
+      "node": {
+        "extensions": [".js", ".jsx"]
+      }
+    }
```
Comment (kilemensi) [inline dominion/frontend/.eslintrc.json]: Why do we need this when we've already configured extensions above in `rules`?

File: dominion/frontend/src/assets/App.css [hunk 3]
```diff
@@ -0,0 +1,94 @@
+.top-menu {
```
Comment (kilemensi) [inline dominion/frontend/src/assets/App.css:3]: Styles should be with their respective components... *If* (and that's a very big if) we need generic `css` file, please explain why and note it in code.

File: dominion/frontend/src/component/Footer/Partners.js [hunk 4]
```diff
@@ -1,15 +1,64 @@
...
-        <Typography variant="h6">Partner&#39;s Logos</Typography>
+    <Grid
+      container
+      spacing={24}
+      direction="row"
+      justify="center"
+      alignItems="center"
+      className={classes.root}
+    >
+      <Grid item lg={2} md={6} sm={6} xs={6} style={{ padding: '2rem' }}>
```
Comment (kilemensi) [inline dominion/frontend/src/component/Footer/Partners.js]: Arrange them from mobile -> desktop and not the other way around i.e Mobile first.

File: dominion/frontend/src/component/Footer/index.js [hunk 5]
```diff
@@ -1,18 +1,188 @@
...
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+    padding: theme.spacing.unit * 3,
+    backgroundImage: `url(${background})`,
+    backgroundRepeat: 'no-repeat',
+    backgroundSize: 'cover',
+    [theme.breakpoints.up('md')]: {
+      height: '50vh'
```
Comment (kilemensi) [inline dominion/frontend/src/component/Footer/index.js]: `v` units means screen size. Here it means the footer needs to be 50% of the screen height. Are you sure you want a footer that high?

File: dominion/frontend/src/component/Header/Header.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import background from '../../assets/images/bg/background.png';
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+    backgroundImage: `url(${background})`,
+    backgroundRepeat: 'no-repeat',
+    backgroundSize: 'cover',
+    [theme.breakpoints.up('md')]: {
+      height: '100vh'
```
Comment (kilemensi) [inline dominion/frontend/src/component/Header/Header.js]: Again, 100% of the screen height should just be the header?

File: dominion/frontend/src/component/Header/Search.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    [theme.breakpoints.up('sm')]: {
+      position: 'absolute',
+      float: 'left'
+    }
+  }
+});
+
+function Search({ classes }) {
+  return (
+    <Grid className={classes.search}>
```
Comment (kilemensi) [inline dominion/frontend/src/component/Header/Search.js]: Why `Grid` here? You are not using it as container and I don't see any `Grid item`. May be a normal `div` could do the job.

PS: For base elements, lets stick to the `classes.root` convention.

**Review by kilemensi — CHANGES_REQUESTED** [review]
There are still merge conflicts @nerdlet.

File: dominion/frontend/src/assets/App.css (hunk 3 above)
Comment (nerdlet) [reply dominion/frontend/src/assets/App.css:3]: @kilemensi Added a comment on why I used css for this

Comment (kilemensi) [reply dominion/frontend/src/assets/App.css:3]: 1. Why do you need custom breakpoints @nerdlet? Material UI does provide breakpoints already.
 2. If you really, really need custom breakpoints, then they should be for the whole app and Material UI again provides a way of doing it via theme [here](https://material-ui.com/customization/default-theme/?expend-path=$.breakpoints.values)

File: dominion/frontend/.eslintrc.json (hunk 1 above)
Comment (kilemensi) [reply dominion/frontend/.eslintrc.json]: All linting issues should be fixed now @nerdlet.

**Review by kilemensi — CHANGES_REQUESTED** [review]
@nerdlet:

 1. `Countries` and `search` from a group that is right aligned. But current implementation doesn't group them and only right aligns the `search`. See design vs implementation below.
screenshot from 2019-02-07 18-16-25
screenshot from 2019-02-07 17-50-10

 2. Click any of the 2 buttons (`Countries` and `search`) should actually show an overlay covering the whole header. The current implementation shows a small dropdown.
screenshot from 2019-02-07 17-49-48
dropdown

**Comment by kilemensi** [comment]
I've moved the Hero PR into this @nerdlet since they're inter-dependent.

**Pull request closed without merging**